### PR TITLE
NR-340296 Update renovate config to update golang version for Dockerfile ARG

### PR DESCRIPTION
**Pull request by sairaj18**
Changes in this PR

Configured renovate.json to update the patch versions for each version in the list `go-version` present in `build_publish_builder_image.yaml` file.

Tested the above by creating a fork of this repo and renovate has created [PR1](https://github.com/sairaj18/coreint-automation-test/pull/9), [PR2](https://github.com/sairaj18/coreint-automation-test/pull/7) (It has created PR1 with automerge true But, I merged the PR1 manually in-order to see if the PR2 gets raised)

### CODE REVIEW

**Review by rajrohanyadav — COMMENTED** [review]
Looks good but one question:
When we add go 1.24 in 3 months, we'll have to update the renovate config as well. Is there a way to achieve this without hardcoding the major go versions?

**Comment by sairaj18** [comment]
> Looks good but one question: When we add go 1.24 in 3 months, we'll have to update the renovate config as well. Is there a way to achieve this without hardcoding the major go versions?

when you add 1.24 manually we need to make sure we add that to the renovate config as well to get the patch versions for 1.24 to auto update. I tried without hardcoding the major go versions but no luck

**Comment by rajrohanyadav** [comment]
> > Looks good but one question: When we add go 1.24 in 3 months, we'll have to update the renovate config as well. Is there a way to achieve this without hardcoding the major go versions?
> 
> when we add 1.24 manually we need to make sure we add that to the renovate config as well to get the patch versions for 1.24 to auto update. I tried without hardcoding the major go versions but no luck

I might try something like this:
```json
{
      "customType": "regex",
      "fileMatch": ["^\\.github/workflows/build_publish_builder_image\\.yaml$"],
      "matchStrings": [
        "go-version:\\s*\\[(?<currentValue>[^\\]]+)]"
      ],
      "datasourceTemplate": "go",
      "versioningTemplate": "semver",
      "depNameTemplate": "go",
      "extractVersionTemplate": "{{#each (split currentValue ',')}}{{this}}{{#unless @last}},{{/unless}}{{/each}}"
}
```

But if it doesn't work. I think it's okay as this will change once every 6 months and it is okay to update the renovate config as well. Unless someone has a better idea.

File: renovate.json
Comment (rajrohanyadav) [inline renovate.json:6]: Do you see a warning about older format? Can we use `customManagers` of `"customType": "regex"`?

**Comment by sairaj18** [comment]
> I might try something like this:
> 
> ```json
> {
>       "customType": "regex",
>       "fileMatch": ["^\\.github/workflows/build_publish_builder_image\\.yaml$"],
>       "matchStrings": [
>         "go-version:\\s*\\[(?<currentValue>[^\\]]+)]"
>       ],
>       "datasourceTemplate": "go",
>       "versioningTemplate": "semver",
>       "depNameTemplate": "go",
>       "extractVersionTemplate": "{{#each (split currentValue ',')}}{{this}}{{#unless @last}},{{/unless}}{{/each}}"
> }
> ```
> 
> But if it doesn't work. I think it's okay as this will change once every 6 months and it is okay to update the renovate config as well. Unless someone has a better idea.

Tried this and also tweaking this config but no luck.

File: renovate.json
Comment (sairaj18) [reply renovate.json]: Done changed `regexManagers` -> `customManagers` and added `"customType": "regex"`

File: renovate.json
Comment (rajrohanyadav) [inline renovate.json:27]: Shouldn't this also have `"customType": "regex"`?

Comment (sairaj18) [reply renovate.json]: yep missed adding here :(, added it now

**Review by rajrohanyadav — APPROVED** [review]
LGTM, please get a second review from AC team.

**Review by DavSanchez — APPROVED** [review]

**Pull request merged**
